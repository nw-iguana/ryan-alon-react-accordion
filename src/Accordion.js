import React, { Component } from 'react'

export class Accordion extends Component {

  static defaultProps = {
    sections: []
  }

    constructor(props) {
        super(props);
        this.state = {
            currentOpenSection: null,
            contentParagraph: null
        }
    }


    handleClick(index) {

        this.setState((prevState) => {
            return {
                currentOpenSection: index
            }
        }, this.renderContent(index))
    }

    renderContent(index) {
        this.setState({
            currentParagraph: (<p>{this.props.sections[index].content}</p>)
        })
    }

    renderButtons() {
        return this.props.sections.map((section, index) => {
            return (
                  <li key={index}>
                      <button onClick={() => this.handleClick(index)}>{section.title}</button>
                  </li>
              )
        })
    }


  render() {
    return (
      <ul>
        {this.renderButtons()}
        {this.state.currentParagraph}
      </ul>
    )
  }
}

export default Accordion
