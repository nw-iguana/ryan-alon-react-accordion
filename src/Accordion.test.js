import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Accordion from './Accordion';


describe('Accordion Component', () => {
    const sections = [
        {
          title: 'Section 1',
          content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
        },
        {
          title: 'Section 2',
          content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
        },
        {
          title: 'Section 3',
          content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
        },
      ]
    
    it('renders no content paragraph by default', () => {
        const tree = renderer.create(<Accordion sections={sections}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('renders an empty li when the sections prop is not supplied', () => {
      const tree = renderer.create(<Accordion />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('the component opens a clicked section', () => {
        const wrapper = shallow(<Accordion sections={sections}/>);
        wrapper.find('button').at(1).simulate('click');
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it('The component only opens the last section when multiple sections have been clicked.', () => {
      const wrapper = shallow(<Accordion sections={sections}/>);
      wrapper.find('button').at(1).simulate('click');
      wrapper.find('button').at(0).simulate('click');
      expect(toJson(wrapper)).toMatchSnapshot();
    })

});