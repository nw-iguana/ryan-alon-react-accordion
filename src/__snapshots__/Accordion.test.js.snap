// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component The component only opens the last section when multiple sections have been clicked. 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit.
  </p>
</ul>
`;

exports[`Accordion Component renders an empty li when the sections prop is not supplied 1`] = `<ul />`;

exports[`Accordion Component renders no content paragraph by default 1`] = `
<ul>
  <li>
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion Component the component opens a clicked section 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
  <p>
    Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
  </p>
</ul>
`;
